cmd/kubeadm/app/cmd/phases/controlplanepreparejoin.go

```diff
@@ -0,0 +1,226 @@
+	}
+}
+
+func newControlPlanePrepareSubPhase(component string) workflow.Phase {
+	return workflow.Phase{
+		Name:         controlPlanePhaseProperties[component].name,
+		Short:        controlPlanePhaseProperties[component].short,
+		Run:          runControlPlaneJoinSubPhase(component),
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func runCertsPhaseLocal(c workflow.RunData) error {
+	data, ok := c.(controlPlanePrepareData)
+	if !ok {
+		return errors.New("certs phase invoked with an invalid data struct")
+	}
+
+	// Skip if this is not a control plane
+	if data.Cfg().ControlPlane == nil {
+		return nil
+	}
+
+	cfg, err := data.InitCfg()
+	if err != nil {
+		return err
+	}
+
+	// Generate missing certificates (if any)
+	return certsphase.CreatePKIAssets(cfg)
+}
+
+func runKubeconfigPhaseLocal(c workflow.RunData) error {
+	data, ok := c.(controlPlanePrepareData)
+	if !ok {
+		return errors.New("kubeconfig phase invoked with an invalid data struct")
+	}
+
+	// Skip if this is not a control plane
+	if data.Cfg().ControlPlane == nil {
+		return nil
+	}
+
+	cfg, err := data.InitCfg()
+	if err != nil {
+		return err
+	}
+
+	// Generate kubeconfig files for controller manager, scheduler and for the admin/kubeadm itself
+	// NB. The kubeconfig file for kubelet will be generated by the TLS bootstrap process in
+	// following steps of the join --experimental-control plane workflow
+	if err := kubeconfigphase.CreateJoinControlPlaneKubeConfigFiles(kubeadmconstants.KubernetesDir, cfg); err != nil {
+		return errors.Wrap(err, "error generating kubeconfig files")
+	}
+
+	return nil
+}
+
+func runControlPlaneJoinSubPhase(component string) func(c workflow.RunData) error {
+	return func(c workflow.RunData) error {
+		data, ok := c.(controlPlanePrepareData)
+		if !ok {
+			return errors.New("control-plane-prepare phase invoked with an invalid data struct")
+		}
+
+		// Skip if this is not a control plane
+		if data.Cfg().ControlPlane == nil {
+			return nil
+		}
+
+		cfg, err := data.InitCfg()
+		if err != nil {
+			return err
+		}
+
+		// Creates static pod manifests file for the control plane components to be deployed on this node
+		// Static pods will be created and managed by the kubelet as soon as it starts
+		fmt.Printf("[control-plane-prepare] Creating static Pod manifest for %q\n", component)
+		return controlplane.CreateStaticPodFiles(kubeadmconstants.GetStaticPodDirectory(), &cfg.ClusterConfiguration, &cfg.LocalAPIEndpoint, component)
+	}
+}
+
+func runCheckLocalEtcdStatus(c workflow.RunData) error {
+	data, ok := c.(controlPlanePrepareData)
+	if !ok {
+		return errors.New("check-local-etcd-status phase invoked with an invalid data struct")
+	}
+
+	// Skip if this is not a control plane
+	if data.Cfg().ControlPlane == nil {
+		return nil
+	}
+
+	cfg, err := data.InitCfg()
+	if err != nil {
+		return err
+	}
+
+	if cfg.Etcd.External != nil {
+		fmt.Println("[control-plane-prepare] Skipping etcd check in external mode")
+		return nil
+	}
+
```

Comment:
still print a message that we are performing an action (preflight)?
`fmt.Println("[preflight] Checking if etcd cluster is healthy")`